src/App.tsx

```diff
@@ -1,39 +1,56 @@
 import React from 'react';
 import './App.scss';
+
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
+import './types/Movie';
 
-type State = {};
+type State = {
+  query: string;
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      query: event.target.value,
+    });
+  };
 
   render() {
+    const { query } = this.state;
+    const queryRegExp = new RegExp(query, 'i');
+
+    const visibleMovies: Movie[] = moviesFromServer.filter(
+      (movie) => queryRegExp.test(movie.title) || queryRegExp.test(movie.description),
+    );
```

Comment:
I don't really see any sense in using regExp, when you could simply use 'includes'. But it's up to you